baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/CollectionsIsEmptyTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.errorprone.CompilationTestHelper;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+
+public class CollectionsIsEmptyTest {
```

Comment:
One edge case we may wish to consider:
Sometimes collections implement
```java
@Override
public boolean isEmpty() {
    return this.size() == 0;
}
```
or without the `this` keyword:
```java
@Override
public boolean isEmpty() {
    return size() == 0;
}
```

Could we special case invocations on the current object and avoid producing changes?

Reply:
Good catch! Will add a test for that.